src/components/ui/editor/VolumeSlider.jsx

```diff
@@ -0,0 +1,58 @@
+import PropTypes from 'prop-types';
+import { useState } from 'react';
+import { HiVolumeOff, HiVolumeUp } from 'react-icons/hi';
+
+import { SOUND_CONFIG } from '@/constants/sound';
+
+const VolumeSlider = ({ label, id, value, onChange }) => {
+  const { DEFAULT_VOLUME } = SOUND_CONFIG.BGM;
+  const { MIN_VOLUME } = SOUND_CONFIG.BGM;
+
+  const [lastVolume, setLastVolume] = useState(value > MIN_VOLUME ? value : DEFAULT_VOLUME);
+
+  const handleMuteToggle = () => {
+    if (value > MIN_VOLUME) {
+      setLastVolume(value);
+      onChange(0);
+    } else {
+      onChange(lastVolume || DEFAULT_VOLUME);
+    }
+  };
```

Comment:
슬라이더와 음소거 버튼을 하나의 컴포넌트로 잘 구성해주신 것 같습니다!

다만 관심사 분리 측면에서 한 가지 제안을 드리고 싶습니다.
현재 음소거 관련 로직이 컴포넌트 내부에 함께 섞여 있어서, `useVolumeToggle`과 같은 커스텀 훅으로 분리하면 더 깔끔하게 관리할 수 있을 것 같아요.

예를 들어 아래처럼 훅을 만들어 사용한다면 `VolumeSlider`는 슬라이더와 버튼만 신경 쓰면 되고, 로직은 별도에서 처리할 수 있어서 나중에 확장이나 유지보수할 때 훨씬 좋을 것 같습니다 ☺️


```suggestion
const { toggle, isMuted } = useVolumeToggle(value);
```